### DISCUSSION TITLE
Custom-Indexer tutorial not populating postgres db with transaction info

### DISCUSSION THREAD

**Discussion by oscarthedev15** [aptos-labs/aptos-developer-discussions#509] [Questions] [answered]
Hi Devs, 
https://aptos.dev/en/build/create-aptos-dapp/templates/custom-indexer
i followed this tutorial and built the dapp with the indexer.  The database migrations work correctly and then the docker image is run, the chain ID is updated.  (so I know the db connection is correct)

What is not working is the db is not being updated for messages when i make transactions on the contract i deployed.  I confirmed the starting version and the contract address is correct in the indexer config.yml file.  My aptos build api key is correct as well. The db is not writing in any table relating to the transactions created.

the front end application is creating messages perfectly fine but they are not populating in the db and therefore not showing up on the ui. 

any and all help is much appreciated!

**Comment by 0x-j** [comment]
Hi thanks for trying out the indexer template, since you can create message successfully on the UI but not show up in DB, this is probably frontend is using a different contract. The contract address used by frontend is `NEXT_PUBLIC_MODULE_ADDRESS` in `.env`, which is created or updated when you deploy contract with `npm run move:publish`. Can you check `NEXT_PUBLIC_MODULE_ADDRESS` is same as `contract_address` in `config.yaml` used by indexer?

And also check frontend is reading from the right db? i.e. `DATABASE_URL` in `.env` is the same as `postgres_connection_string` in `config.yaml` of indexer?

**Reply by oscarthedev15** [reply]
I can confirm the contract in .env is the same as the contract in the config.yaml file. So I do not think that is the problem. This is also verified from the transactions the front end is creating.

**Reply by 0x-j** [reply]
are you running indexer locally or on the cloud? and is the `last_success_version` in `processor_status` table being constantly updated when you run the indexer? Want to make sure indexer is not panic

**Reply by oscarthedev15** [reply]
running locally

I can confirm the last_success_version is being updated constantly when i run the indexer image

processor_status table gets updated
ledger_infos table gets populated with chain_id, 2 when the docker image is run 

no other tables get populated. The docker logs run continuously logging like this:
{"timestamp":"2024-10-26T22:13:49.004118Z","level":"INFO","message":"[Transaction Stream] Received transactions from GRPC.","stream_address":"https://grpc.testnet.aptoslabs.com/","connection_id":"965449cd-ed5f-42a9-8441-0710caa60422","start_version":6184867990,"end_version":6184867994,"start_txn_timestamp_iso":"2024-10-26T22:13:48.588846000Z","end_txn_timestamp_iso":"2024-10-26T22:13:48.588846000Z","num_of_transactions":5,"size_in_bytes":28618,"duration_in_secs":0.123761834,"tps":41,"bytes_per_sec":231234.4531028847,"filename":"/usr/local/cargo/git/checkouts/aptos-indexer-processor-sdk-2f3940a333c8389d/ca658a0/rust/transaction-stream/src/transaction_stream.rs","line_number":403,"threadName":"tokio-runtime-worker","threadId":"ThreadId(12)"}

**Reply by 0x-j** [reply]
can you upload everything to a github repo? also include the config and .env, you can set stuff like auth token to empty, i want to run it on my end

**Reply by oscarthedev15** [reply]
https://github.com/oscarthedev15/aptos-indexer

here is the repo.  I removed only the db connection and the auth token from the config, if you want the db connection I am happy to also provide.

**Reply by 0x-j** [reply, marked as answer]
im able to reproduce, i fixed it in this [PR](https://github.com/aptos-labs/create-aptos-dapp/pull/260), we have already released a new version on npm, 0.0.36. Can you try creating a project using latest version? `npx create-aptos-dapp@latest`

**Reply by 0x-j** [reply]
very very appreciate you trying this template out and catching the error 🫡

**Reply by oscarthedev15** [reply]
thank you sir for the swift fix.

**Reply by oscarthedev15** [reply]
hi sir,
quick question but is there any way to speed up indexing? 
i am re-indexing my db with a module deployed a few weeks ago and it understandably is taking a while to re-process all transactions.  I am wondering is there a way to speed this up?

**Reply by 0x-j** [reply]
hey did you set the [starting_version](https://github.com/aptos-labs/create-aptos-dapp/blob/main/templates/custom-indexer-template/indexer/example.config.yaml#L9) to the version that your contract was deployed? If you start indexing from a few weeks ago, it should take a couple hours at most I think
